Merge `dev` into `feat-fast-sync`.

afck opened the pull request:
Closes https://github.com/casper-network/casper-node/issues/2042.

Take a look at https://github.com/casper-network/casper-node/pull/2266 to see the changes compared to `dev` instead of `feat-fast-sync`.

Commit:
Regression test

Commit:
Fix

Fraser999 committed:
update feature sets on casper-types and casper-contract crates

casperlabs-bors-ng[bot] committed:
Merge #2090

2090: Revert "GH-2079: Bump transfer to 250m gas" r=mpapierski a=mpapierski

Reverts casper-network/casper-node#2084

We'll postpone this change, and we're analyzing this from an economic perspective

Co-authored-by: Michał Papierski <michal@papierski.net>

Commit:
1977: Fix naming error in CHANGELOG

Commit:
update rpc_schema_hashing_V(n).json

casperlabs-bors-ng[bot] committed:
Merge #2092

2092: NO-TICKET: Improve logs consistency. r=goral09 a=goral09

Closes https://github.com/casper-network/casper-node/issues/2093

Co-authored-by: Mateusz Górski <gorski.mateusz@protonmail.ch>

Fraser999 committed:
stop silently ignoring parse errors for complex runtime args in client

casperlabs-bors-ng[bot] committed:
Merge #2095

2095: Stop silently ignoring parse errors for complex runtime args in client r=Fraser999 a=Fraser999

The client currently silently ignores errors which occur when passing a value via `--session-args-complex` or `--payment-args-complex`.  This PR fixes that, causing the client to fail and report the error to the user.  It also adds tests for these two cases.

Closes #2094.


Co-authored-by: Fraser Hutchison <fraser@casperlabs.io>

luis-casper committed:
Implementing code comment fixes.

I am also adding a few TODOs that need some attention from people that
are more familiar with the EE.

Fraser999 committed:
rename read_trie to get_trie

Commit:
Merge pull request #13 from Fraser999/retrieve-state-rpc-and-tool

Rename read_trie to get_trie

casperlabs-bors-ng[bot] committed:
Merge #2091

2091: Enable the casper-mainnet feature by default in manifests r=fizyk20 a=fizyk20

This adds the `casper-mainnet` feature to the set of default features in the node and the client. This feature is responsible for keeping the software on the legacy hashing scheme in protocol versions up to the one where we will switch to fast sync, and should be enabled in all builds intended for use in the mainnet.

Co-authored-by: Bartłomiej Kamiński <bart@casperlabs.io>

casperlabs-bors-ng[bot] committed:
Merge #2078

2078: Add `read_trie` rpc and state download tool r=Fraser999 a=dwerner

Resolves part of #1993 (1/2)

The purpose of this PR is to:
- Add a JSON-RPC endpoint to read a trie as a base16 encoded blob.
- Add a state downloader that will use this endpoint to download the trie, and also any blocks+deploys.

Replaces https://github.com/casper-network/casper-node/pull/1994

Co-authored-by: Daniel Werner <dan@casperlabs.io>
Co-authored-by: Fraser Hutchison <fraser@casperlabs.io>
Co-authored-by: Daniel N. Werner <dan@casperlabs.io>

Commit:
1977: Fix naming error in CHANGELOG

Commit:
1977: Ensure consistency with `validator_changes`

Commit:
1977: Resolve merge conflicts

Fraser999 committed:
Merge remote-tracking branch 'upstream/dev' into 2088-no-std-features

Fraser999 committed:
update retrieve-state tool to match other crate updates

Commit:
set enable_manual_sync to false by default

casperlabs-bors-ng[bot] committed:
Merge #2099

2099: Set enable_manual_sync to false by default r=dwerner a=dwerner

As discussed in sprint planning, this PR sets enable_manual_sync to false by default.


Co-authored-by: Daniel Werner <dan@casperlabs.io>

Commit:
1977: Add tests to ValidatorChange

Commit:
dry-run-deploys feature

Commit:
removing use of deprecated code.

Commit:
refactoring tests that used deprecated code.

fizyk20 committed:
Linear Chain Sync: upgrade or downgrade on version mismatch

Fraser999 committed:
apply json-schema feature

casperlabs-bors-ng[bot] committed:
Merge #2096

2096: Update feature sets on casper-types and casper-contract crates r=Fraser999 a=Fraser999

This PR changes the `casper-types` and `casper-contract` to be `no_std` by default.

The types crate has individual features to enable support for `datasize`, `schemars` and `proptest`.  If any of these are enabled, the std library is included, but it's anticipated that smart contract authors will never need to enable these features.

Using the code in this PR locally I checked that the full set of tests run via `make check-rs` is unchanged.

The PR also adds `codegen-units = 1` and `lto = true` to the release profile in order to reduce the size of the Wasm contracts.  Due to our choice to use a single workspace, these settings are applied to all targets, even though they're only really useful for some of the targets.  However, it didn't significantly affect the overall time to execute `make check-rs` and it did reduce the size of the compiled Wasm from 5.3MB to 3.2MB.

Closes #2088.

Co-authored-by: Fraser Hutchison <fraser@casperlabs.io>

Commit:
2085: Address PR feedback pt.1

luis-casper committed:
Fixing some TODOs.

luis-casper committed:
Converting MS-DOS end-of-line to UNIX.

Fraser999 committed:
cleanups relating to get-validator-changes

fizyk20 committed:
Address a review comment

Fraser999 committed:
address further get-validator-changes nitpicks

Commit:
Clarify comment on precomputed payment errors.

Fraser999 committed:
detect cases of unplanned shutdowns when an upgrade is due

Commit:
Merge branch 'dev' into gh-1764-core-docs-1

 Conflicts:
	execution_engine/src/core/engine_state/balance.rs
	execution_engine/src/core/engine_state/era_validators.rs
	execution_engine/src/core/engine_state/error.rs
	execution_engine/src/core/engine_state/execute_request.rs
	execution_engine/src/core/engine_state/genesis.rs
	execution_engine/src/core/engine_state/mod.rs
	execution_engine/src/core/engine_state/op.rs
	execution_engine/src/core/engine_state/query.rs
	execution_engine/src/core/engine_state/run_genesis_request.rs
	execution_engine/src/core/engine_state/step.rs
	execution_engine/src/core/engine_state/upgrade.rs
	execution_engine/src/core/execution/address_generator.rs
	node/src/components/rpc_server/rpcs/state.rs
	smart_contracts/contract/src/contract_api/mod.rs
	types/src/cl_type.rs
	types/src/contracts.rs

darthsiroftardis committed:
Merge pull request #12 from Fraser999/1977-get-validator-changes

Cleanups relating to get-validator-changes

casperlabs-bors-ng[bot] committed:
Merge #2086

2086: Wire up Consensus information to JSON-RPC and REST  r=darthsiroftardis a=darthsiroftardis

CHANGELOG:

- Added a new JSON-RPC endpoint `get-validator-info` as well as a new REST endpoint `validator-changes` that reports the changes in the validator set between two eras.
- Added a new command to the client binary `get-validator-info` 
- Wired up the new command the client library FFI
- Added a test to the client integration tests for the new command

Closes: #1977 

Co-authored-by: Andreas Fackler <andreas@casperlabs.io>
Co-authored-by: casperlabs-bors-ng[bot] <82463608+casperlabs-bors-ng[bot]@users.noreply.github.com>
Co-authored-by: Karan Dhareshwar <karan@casperlabs.io>
Co-authored-by: Fraser Hutchison <fraser@casperlabs.io>
Co-authored-by: Karan Dhareshwar <42871449+darthsiroftardis@users.noreply.github.com>

Commit:
Fix JSON fixtures after changing schema.

luis-casper committed:
Documenting undefined behavior.

Commit:
Fix one instance of unchecked arithmetic.

Fraser999 committed:
update cargo-casper readme

Commit:
Shared module rustdocs

casperlabs-bors-ng[bot] committed:
Merge #2105

2105: Update cargo-casper readme r=Fraser999 a=Fraser999

This PR updates the cargo-casper README file, including details of the `--erc20` option.

Closes #2104.

Co-authored-by: Fraser Hutchison <fraser@casperlabs.io>

Commit:
2085: Address PR feedback pt.2

Commit:
2085: Address PR feedback pt.2

luis-casper committed:
Fixing code comments.

sacherjj committed:
Updated to chainspec.toml for block speed increase improvements.
Reduced deploy_delay from 1min to 15sec.

sacherjj committed:
Updating maximum_round_exp to 17 from 18

Commit:
2085: Drop clone and amend fetcher test

Fraser999 committed:
resolve duplicated max_associated_keys config option

casperlabs-bors-ng[bot] committed:
Merge #2108

2108: Resolve duplicated max_associated_keys config option r=Fraser999 a=Fraser999

This PR deduplicates the `max_associated_keys` config option.

Closes [#73](https://github.com/casper-network/Governance/issues/73).

Co-authored-by: Fraser Hutchison <fraser@casperlabs.io>

Commit:
Add changelog entry.

Commit:
Fix missing directives

casperlabs-bors-ng[bot] committed:
Merge #1854

1854: GH-1764: Add docs to the core modules. r=mpapierski a=mpapierski

Ref #1764 

This PR adds rust docs to the `core` module and its children.


Co-authored-by: Michał Papierski <michal@casperlabs.io>
Co-authored-by: Michał Papierski <michal@papierski.net>
Co-authored-by: Luís Fernando Schultz Xavier da Silveira <luis@casperlabs.io>

Commit:
Merge remote-tracking branch 'upstream/dev' into DA-work

# Conflicts:
#	execution_engine/src/core/engine_state/executable_deploy_item.rs

sacherjj committed:
Adding shutdown_on_standstill flag back in.

casperlabs-bors-ng[bot] committed:
Merge #2081

2081: Expand Deploy Acceptor r=darthsiroftardis a=darthsiroftardis

Co-authored-by: Ed Hastings <ed@casperlabs.io>

CHANGELOG:

- Expanded the checks the `DeployAcceptor` makes on a single `Deploy`
     - Account related checks
          1. Checks if the `AccountHashes` associated with the `Deploy` are valid  associated keys
          2. Checks if the `Weight` of the `AccountHashes` meet the threshold to perform a deployment
          3. Checks if the balance of the `Account` in the case of a `Deploy` received by from a `Client` is sufficient 
     - ExecutableDeployItem
         1. Checks if the ExecutableDeployItem present in the session or payment fields are valid and present in the case of stored contracts and contract packages
 - Moved the cryptographic validity check of a `Deploy` to be the last step once all other prerequisites are met.
 - Expanded the test suite to cover the newly added checks
         
Closes #2085 


Co-authored-by: Karan Dhareshwar <karan@casperlabs.io>

Commit:
Fix formatting.

sacherjj committed:
Removing shutdown_on_standstill flag as is true when standstill_timeout exists.

casperlabs-bors-ng[bot] committed:
Merge #2107

2107: Updated to chainspec.toml and config for block speed increase improvements. r=sacherjj a=sacherjj

Chainspec.toml changes for exp 15 (32s) blocks and seniorage calcs
Reducing deploy_delay from 1min to 15sec for default in config


Co-authored-by: Joe Sacher <321623+sacherjj@users.noreply.github.com>

mpapierski committed:
Apply suggestions from code review

Co-authored-by: Fraser Hutchison <190532+Fraser999@users.noreply.github.com>

Commit:
Address @Fraser999's review comments

Fraser999 committed:
remove incorrect validity check from deploy acceptor

Commit:
Clarify comment on determinism of transforms.

Co-authored-by: Fraser Hutchison <fraser@casperlabs.io>

luis-casper committed:
Fixing grammar mistakes.

fizyk20 committed:
Add log messages when upgrading/downgrading in Linear Chain Sync

Commit:
remove checksum_encode_if_small; update docs.

Commit:
2120: Fix bug in Deploy acceptor

casperlabs-bors-ng[bot] committed:
Merge #2101

2101: Linear Chain Sync: upgrade or downgrade on version mismatch r=fizyk20 a=fizyk20

When the Linear Chain Sync component attempts to execute a block and the block's protocol version differs from its own, it will upgrade/downgrade to match the versions.

This should prevent issues when a node that missed an upgrade initially is being upgraded with a delay, with a trusted hash from after the upgrade. (The node would then execute a block with a non-matching protocol version and exit with a fatal error because the executed block wouldn't match the downloaded one.)


Co-authored-by: Bartłomiej Kamiński <bart@casperlabs.io>
Co-authored-by: Fraser Hutchison <fraser@casperlabs.io>

Commit:
Clarify commutativity properties of Transform.

Commit:
2120: Remove redundant errors in DA

Commit:
2120: PR prep

Commit:
2120: Add tests for payment amount

Commit:
2120: Address PR feedback pt.1

nilz3ro committed:
merge dev into cep-57-edit-2

casperlabs-bors-ng[bot] committed:
Merge #2110

2110: GH-1765: Add docs to the shared module. r=mpapierski a=mpapierski

Closes #1765

This PR adds rust docs to the shared module in EE and its children.

Co-authored-by: Michał Papierski <michal@casperlabs.io>
Co-authored-by: Luís Fernando Schultz Xavier da Silveira <luis@casperlabs.io>
Co-authored-by: Michał Papierski <michal@papierski.net>
Co-authored-by: luis-casper <90623863+luis-casper@users.noreply.github.com>

casperlabs-bors-ng[bot] committed:
Merge #2126

2126: Fix bug in Deploy acceptor r=darthsiroftardis a=darthsiroftardis

CHANGELOG:

- Fixed a bug in the deploy acceptor that bypassed checks on deploys using standard payment.
- Removed two redundant errors in `DeployParameterFailure`.
- Added 5 tests around the deploy acceptor to check for arguments in native transfer and standard payment.
- Added tests in deploy to check for arguments in native transfer.

Closes #2120 

Co-authored-by: Fraser Hutchison <fraser@casperlabs.io>
Co-authored-by: Karan Dhareshwar <karan@casperlabs.io>

Commit:
update test fixtures; error handling.

Commit:
2128: Change native transfer cost

Commit:
2128: Reflect cost change in testing chainspecs

casperlabs-bors-ng[bot] committed:
Merge #2129

2129: Increase price of wasmless transfer r=darthsiroftardis a=darthsiroftardis

CHANGELOG

- Increased price of wasmless (native) transfer cost to 100 million motes (0.1 CSPR)

Closes #2128 

Co-authored-by: Karan Dhareshwar <karan@casperlabs.io>

Commit:
2130: Peform account checks on client deploys only

Commit:
Add regression test

Commit:
wip fix

Commit:
2130: Remove account existence check

Commit:
Comment out zero gas payment result check

Commit:
Fix failling tests

Co-authored-by: Karan Dhareshwar <karan@casperlabs.io>

Fraser999 committed:
update deploy acceptor tests

Commit:
Fix for zero cost payment code and cleanups

Commit:
Clean up commented out code

darthsiroftardis committed:
Merge pull request #13 from Fraser999/2130-fix-deploy-acceptor

Update deploy acceptor tests

casperlabs-bors-ng[bot] committed:
Merge #2131

2131: Perform account checks on deploys from clients only. r=darthsiroftardis a=darthsiroftardis

CHANGELOG:

- Changed the behavior of the deploy acceptor to never perform account checks on deploys sent by peers.
- Added two tests to ensure account checks for deploys sent by peers don't occur


Closes #2130 

Co-authored-by: Karan Dhareshwar <karan@casperlabs.io>
Co-authored-by: Fraser Hutchison <fraser@casperlabs.io>
Co-authored-by: Karan Dhareshwar <42871449+darthsiroftardis@users.noreply.github.com>

mpapierski committed:
Update execution_engine/src/core/engine_state/mod.rs

Co-authored-by: Fraser Hutchison <190532+Fraser999@users.noreply.github.com>

mpapierski committed:
Update execution_engine_testing/tests/src/test/regression/regression_20210924.rs

Co-authored-by: Fraser Hutchison <190532+Fraser999@users.noreply.github.com>

Commit:
Fix test

Commit:
Revert duplicated check for 0 gas payment.

This duplicates logic in check_forced_transfer where we check if payment
purse balance < max_payment, and it still makes the regression test
pass.

Commit:
Rustfmt

casperlabs-bors-ng[bot] committed:
Merge #2132

2132: Fix minimum penalty charge r=mpapierski a=mpapierski

Co-authored-by: Ed Hastings <ed@casperlabs.io>
Co-authored-by: Fraser Hutchison <fraser@casperlabs.io>
Co-authored-by: Karan Dhareshwar <karan@casperlabs.io>


Co-authored-by: Michał Papierski <michal@casperlabs.io>
Co-authored-by: Michał Papierski <michal@papierski.net>

afck committed:
Improve and add logging to run_equivocator_network test.

The test is flaky on CI but always passes locally. This adds some
logging so we can see whether it's a timing issue.

We also start isolating one of the two clone nodes for three rounds
starting from the first _consensus_ message it sends now, instead of the
first message it receives: If other messages are exchanged before the genesis
timestamp (address gossiping?), the three round lengths could expire
before the first consensus message is even sent.

nilz3ro committed:
Update types/src/checksummed_hex.rs

Co-authored-by: Matthew Doty <matthew.wampler.doty@gmail.com>

nilz3ro committed:
Update types/src/checksummed_hex.rs

Co-authored-by: Matthew Doty <matthew.wampler.doty@gmail.com>

nilz3ro committed:
Update types/src/checksummed_hex.rs

Co-authored-by: Matthew Doty <matthew.wampler.doty@gmail.com>

Commit:
remove debug code.

Commit:
Merge branch 'cep-57-edit-2' of github.com:nilz3ro/casper-node into cep-57-edit-2

Commit:
CheckSummed -> Checksummed.

afck committed:
Rename first_message to first_message_time.

Commit:
check if char is included in range;

Commit:
use const for ranges; use char syntax.

Commit:
use byte literal syntax.

Commit:
Copy over changes after review.

afck committed:
Pass RNG into consensus protocol instead of using thread_rng.

afck committed:
Avoid some unnecessary synchronizer messages.

If we have unit number `n`, we should request units starting from
 `n + 1`, not `n`. And if the peer has unit number `n`, we should
start sending them units starting from `n + 1`.

Commit:
removing error type.

afck committed:
Deduplicate find_in_swimlane/find_ancestor_unit.

afck committed:
Fix panorama mismatch error log.

afck committed:
Rename Count to NextSeq.

Commit:
allow deprecated code in source files.

Commit:
2152: Remove system contract cache

Commit:
remove Code and Value.

Commit:
use storage component directly

Commit:
fix doc tests and integration test.

nilz3ro committed:
merge dev into GH-1978

Commit:
Merge branch 'dev' into cep-57-edit-2

Commit:
updates based on pr feedback.

Commit:
obey clippy.

Commit:
add bench config to toml.

casperlabs-bors-ng[bot] committed:
Merge #2178

2178: Check the immediate caller in system contracts. r=sacherjj a=mpapierski

Checks if immediate caller is of session type of the same account as the provided account hash.

Co-authored-by: Michał Papierski <michal@casperlabs.io>

casperlabs-bors-ng[bot] committed:
Merge #2133

2133: Improve and add logging to run_equivocator_network test. r=afck a=afck

The test is flaky on CI but always passes locally. This adds some logging so we can see whether it's a timing issue.

We also start isolating one of the two clone nodes for three rounds starting from the first _consensus_ message it sends now, instead of the first message it receives: If other messages are exchanged before the genesis timestamp (address gossiping?), the three round lengths could expire before the first consensus message is even sent.

See issue #1859.

Co-authored-by: Andreas Fackler <andreas@casperlabs.io>

afck committed:
Merge remote-tracking branch 'upstream/dev' into highway_rng

casperlabs-bors-ng[bot] committed:
Merge #2167

2167: Highway/synchronizer: pass in RNG, avoid unnecessary requests, deduplicate. r=afck a=afck

This addresses a few minor issues from https://github.com/casper-network/casper-node/pull/2062:
* The RNG is passed into the Highway protocol implementation instead of using `thread_rng`, to be in line with other modules.
* Requests for missing units start at `n + 1`, not `n`, if our latest unit is `n`.
* Deduplication of `find_in_swimlane` and `find_ancestor_unit`, which were equivalent.

Closes https://github.com/casper-network/casper-node/issues/2061

Co-authored-by: Andreas Fackler <andreas@casperlabs.io>

TomVasile committed:
remove vault dependency

Commit:
tweaks to make rust-analyzer happy

Commit:
use tempfile::tempdir over testdir

Commit:
revert cargo.lock

Commit:
Update execution_engine/src/storage/trie_store/lmdb.rs

Co-authored-by: Matthew Doty <matthew.wampler.doty@gmail.com>

Commit:
Update types/src/bytesrepr.rs

Co-authored-by: Matthew Doty <matthew.wampler.doty@gmail.com>

Commit:
Update utils/retrieve-state/src/storage.rs

Co-authored-by: Matthew Doty <matthew.wampler.doty@gmail.com>

Commit:
fix rpc schema test

Commit:
Merge branch 'dry-run-deploys-dev' of ssh://github.com/dwerner/casper-node into dry-run-deploys-dev

TomVasile committed:
changing where to look for data.lmdb

TomVasile committed:
fix whitespace

TomVasile committed:
remove debug echo

Commit:
Fix generation of global state update

casperlabs-bors-ng[bot] committed:
Merge #2198

2198: No tick nctl storage path fix r=EdHastingsCasperLabs a=TomVasile

Changes:
- moves path for data.lmdb for version >= 1.4.0
- Updates the functionailty for `global-state-update-gen`, where `system-contract-registry` takes an optional argument `state-hash` which fetches system contract hashes from the trie store as opposed to the protocol data DB


Co-authored-by: TomVasile <43349666+TomVasile@users.noreply.github.com>
Co-authored-by: Karan Dhareshwar <karan@casperlabs.io>

casperlabs-bors-ng[bot] committed:
Merge #2191

2191: CI: remove vault dependency [dev] r=TomVasile a=TomVasile

Changes:
- Adjusts wrapper and drone file to not use vault.

Issue: https://app.zenhub.com/workspaces/cn-sre-60d363546afb85000e491ff6/issues/casper-network/sre/155

Failed Build: https://drone-auto-casper-network.casperlabs.io/casper-network/casper-node/2491/4/8

Co-authored-by: TomVasile <43349666+TomVasile@users.noreply.github.com>

TomVasile committed:
port https://github.com/casper-network/casper-node/commit/dcab3f0f56ff943003482a1f8d6c020c886f7a84 from 1.4.0 -> dev

casperlabs-bors-ng[bot] committed:
Merge #2020

2020: NCTL: Post-test health checks r=TomVasile a=TomVasile

### Changes:
- Adds `health_checks.sh` to the end of each scenario run in the nightly tests.
- `health_checks.sh` checks the following and compares against allowed values. (Ignorable in case of variable counts or issues.)
    - equivocators
    - crashes
    - restarts
    - doppels
    - errors
    - ejections
- Adjusts `do_start_node` in `itst.sh` to delete pid files. This is due to stopping and starting nodes in NCTL falsely raising a crash in the logs.

### Ticket:
- https://app.zenhub.com/workspaces/cn-sre-60d363546afb85000e491ff6/issues/casper-network/sre/42

### Related:
**Some checks are set to ignore currently due to some unexpected errors in logs while working on this. I ticketed them for follow-up. (Note: I've yet to look at these, just captured to not forget.)**

- https://app.zenhub.com/workspaces/cn-sre-60d363546afb85000e491ff6/issues/casper-network/sre/71
- https://app.zenhub.com/workspaces/cn-sre-60d363546afb85000e491ff6/issues/casper-network/sre/72
- https://app.zenhub.com/workspaces/cn-sre-60d363546afb85000e491ff6/issues/casper-network/sre/77
- https://app.zenhub.com/workspaces/cn-sre-60d363546afb85000e491ff6/issues/casper-network/sre/78
- https://app.zenhub.com/workspaces/cn-sre-60d363546afb85000e491ff6/issues/casper-network/sre/79

Co-authored-by: TomVasile <43349666+TomVasile@users.noreply.github.com>

casperlabs-bors-ng[bot] committed:
Merge #2182 #2199

2182: dry-run-deploys tool r=dwerner a=dwerner

Resolves issues: #2097, #2121 

Brings in dry-run-deploys cli tool.

Replaces and combines old PRs #2168 and #2102

2199: CI: port dcab3f0 from 1.4.0 -> dev r=TomVasile a=TomVasile

Changes:
- removes `s3-genesis-rel-pr` from `.drone.yml`

1.4.0 change: https://github.com/casper-network/casper-node/commit/dcab3f0f56ff943003482a1f8d6c020c886f7a84

Co-authored-by: Daniel Werner <dan@casperlabs.io>
Co-authored-by: Daniel N. Werner <dan@casperlabs.io>
Co-authored-by: TomVasile <43349666+TomVasile@users.noreply.github.com>

casperlabs-bors-ng[bot] committed:
Merge #2193

2193: tweaks to make rust-analyzer happy r=dwerner a=dwerner

resolves #1952 

Co-authored-by: Daniel Werner <dan@casperlabs.io>

Commit:
sync latest from dev.

Commit:
merge and fix conflicts.

Commit:
add doc comments for deploy_item_builder.rs

Commit:
add doc comments for execute_request_builder.rs

Commit:
add basic documentation to lib.rs and utils.rs

Commit:
adding doc comments for wasm_test_builder.rs

Commit:
add doc comments for wasm_test_builder.rs

Commit:
removing TestContext and SessionBuilder.

Commit:
remove unused code & refactor tests.

Commit:
delete test_context.rs and session.rs + fix lints.

Commit:
fix doc comments.

Commit:
remove unused dep.

Fraser999 committed:
improve performance of checksummed hex

Commit:
Add the `schema_status()` test

Commit:
Add the `schema_validator_changes()` test

Commit:
Add the `schema_rpc_schema()` test

Commit:
Deduplicate code by extracting the `assert_schema()` function

Commit:
fix names.

afck committed:
Merge remote-tracking branch 'upstream/dev' into fast-sync-merge

Fraser999 committed:
reinstate behavior of complex args parsing in client

nilz3ro committed:
Merge pull request #3 from Fraser999/checksummed-hex

improve performance of checksummed hex

Commit:
2152: Update CHANGELOG

casperlabs-bors-ng[bot] committed:
Merge #2176

2176: Remove system contract cache r=darthsiroftardis a=darthsiroftardis

CHANGELOG:

- Removed `SystemContractCache` as it was not being used anymore


Closes #2152 

Co-authored-by: Karan Dhareshwar <karan@casperlabs.io>

Commit:
update integration test.

Fraser999 committed:
fix chainspec loader test and add a further test

Fraser999 committed:
change client complex args to use checksummed hex decoding

afck committed:
Fix the should_fetch_from_peer test after merge.

The deploy acceptor now sets a deploy's `is_valid` flag, so the deploys
in the test must also have it for the equality test.

afck committed:
Fix run_equivocator_network test.

We keep fewer eras in memory now, and have a genesis block that is the
only block in era 0, so the test needs to be updated.

Commit:
update doc comments.

Commit:
fix naming.

nilz3ro committed:
Update execution_engine_testing/test_support/src/lib.rs

Co-authored-by: Fraser Hutchison <190532+Fraser999@users.noreply.github.com>

nilz3ro committed:
Update execution_engine_testing/test_support/src/lib.rs

Co-authored-by: Fraser Hutchison <190532+Fraser999@users.noreply.github.com>

Commit:
fixes based on PR feedback.

Commit:
Merge branch 'GH-1978' of github.com:nilz3ro/casper-node into GH-1978

Commit:
merge latest dev into GH-1978.

Commit:
fix merge conflicts.

nilz3ro committed:
Merge pull request #4 from Fraser999/checksummed-hex

change client complex args to use checksummed hex decoding

Commit:
document step_request_builder.rs.

casperlabs-bors-ng[bot] committed:
Merge #2211

2211: fix chainspec loader test and add a further test r=Fraser999 a=Fraser999

This PR fixes a chainspec loader test which should have been updated as part of #2101.

It also adds a further test case which explicitly exercises the new code path introduced in that same PR and which was causing the other test to spuriously fail.

Closes #2210.


Co-authored-by: Fraser Hutchison <fraser@casperlabs.io>

Fraser999 committed:
minor updates related to checksummed-hex encoding

nilz3ro committed:
Merge pull request #5 from Fraser999/checksummed-hex

minor updates related to checksummed-hex encoding

Commit:
add comment.

Commit:
fixing digest deserialization.

afck committed:
Fix RPC test; simplify fetcher test fix.

Commit:
Add ignore RUSTSEC-2020-0071 to Makefile

casperlabs-bors-ng[bot] committed:
Merge #2219

2219: Add ignore RUSTSEC-2020-0071 to Makefile r=darthsiroftardis a=darthsiroftardis

CHANGELOG:

- Add ignore RUSTSEC-2020-0071 to Makefile

Closes #2220 


Co-authored-by: Karan Dhareshwar <karan@casperlabs.io>

afck committed:
Only upgrade after making sure the switch block was stored.

Commit:
Add ignore for RUSTSEC-2020-0159

casperlabs-bors-ng[bot] committed:
Merge #2229

2229: Add ignore for RUSTSEC-2020-0159 r=darthsiroftardis a=darthsiroftardis

CHANGELOG:

- Added an ignore for RUSTSEC-2020-0159 to the audit command in the `Makefile`


Closes: #2228 


Co-authored-by: Karan Dhareshwar <karan@casperlabs.io>

Commit:
Merge branch 'dev' into cep-57-edit-2

afck committed:
Test storing block before upgrade.

afck committed:
Remove chainspec from storage.

casperlabs-bors-ng[bot] committed:
Merge #2222

2222: Only upgrade after making sure the switch block was stored. r=afck a=afck

The era supervisor stops voting in an era once the switch block has remained finalized without new equivocations occurring for a while, to make sure all other nodes can verify its finality in the consensus protocol. Once it stops voting it also checks whether it was the last era before an upgrade, and if that is the case, it shuts down the node so that the next software version can take over.

In extreme cases where block execution is delayed by a lot this could potentially happen before the block was executed and stored.

This fix moves the trigger for era deactivation to `handle_block_added`, which is only called after the block has been stored.

Closes #2209.

Co-authored-by: Andreas Fackler <andreas@casperlabs.io>

Commit:
Revert ERC20 in cargo-casper

Commit:
2237: PR prep and cleanup

Commit:
2237: PR prep and cleanup

TomVasile committed:
changes for mixed-case hex

nilz3ro committed:
Merge pull request #6 from TomVasile/mixed_case_fix

changes for mixed-case hex

Fraser999 committed:
remove simple subdir from cargo-casper

afck committed:
Bring contract runtime more in line with dev.

afck committed:
Reverting a few unneeded changes compared to dev.

afck committed:
Rename blake_hashes to hashes.

afck committed:
Merge remote-tracking branch 'upstream/dev' into fast-sync-merge

afck committed:
Apply rustfmt.

afck committed:
Apply right rustfmt version.

afck committed:
Apply rustfmt again.

casperlabs-bors-ng[bot] committed:
Merge #2212

2212: Add REST server schema test r=rafal-ch a=rafal-ch

This PR adds tests for checking if the Json responses returned from the following REST API paths respect the expected schema:
* `/status`
* `/validator-changes`
* `/rpc-schema`

Closes #2145.

Co-authored-by: Rafał Chabowski <rafal@casperlabs.io>

xcthulhu commented on .drone.yml:
```diff
@@ -9,6 +9,13 @@ name: pre-checks
+    # - name: markdown-link-check
+    #   image: casperlabs/node-build-u1804
+    #   commands:
+    #     - "./ci/markdown_link_check.sh"
```
What happened here?

afck replied:
I don't know: https://github.com/casper-network/casper-node/commit/4e31be1c587220d6ebfffe04f16c22332b091070

xcthulhu commented on client/lib/ffi.rs:
```diff
@@ -25,7 +25,7 @@ fn set_last_error(error: Error) {
 ///
 /// The full error can be extracted with get_last_error. See [Error](super::Error) for more details
 /// on what these mean.
-#[allow(non_snake_case, non_camel_case_types, missing_docs)]
```
I'm curious why we are changing this here in the ffi...?

xcthulhu commented on execution_engine/CHANGELOG.md:
```diff
@@ -15,18 +15,26 @@ All notable changes to this project will be documented in this file.  The format
+* Added a support for passing a public key as a `target` argument in native transfers.
+* Added a `max_associated_keys` configuration option for a hard limit of associated keys under accounts.
+
+
```
Few too many whitespaces here

afck replied:
Would be better to fix this on `dev`, not `feat-fast-sync`. We didn't change this file.

Commit:
fix asymmetric type to_hex and from_hex.

xcthulhu commented on execution_engine/src/core/engine_state/executable_deploy_item.rs:
```diff
@@ -1,3 +1,4 @@
+//! Units of execution.
 // TODO - remove once schemars stops causing warning.
 #![allow(clippy::field_reassign_with_default)]
```
Can this be removed?

afck replied:
Also better on `dev` maybe.

afck committed:
Remove explicit compile_error.

afck committed:
Fix rustdoc.

Commit:
update tests.

Commit:
Remove superfluous return

casperlabs-bors-ng[bot] committed:
Merge #2246

2246: Backport 'removal of ERC20 support from cargo-casper' to dev r=darthsiroftardis a=darthsiroftardis

CHANGELOG:

- Backport of the [PR 2238](https://github.com/casper-network/casper-node/pull/2238)

Originally closed #2237 

Co-authored-by: Karan Dhareshwar <karan@casperlabs.io>
Co-authored-by: Fraser Hutchison <fraser@casperlabs.io>

Commit:
Merge branch 'dev' into cep-57-edit-2

goral09 committed:
Do not crash node when AuctionState query can't be answered.

Commit:
update json schema.

casperlabs-bors-ng[bot] committed:
Merge #2179

2179: Checksummed hex r=nilz3ro a=nilz3ro

# About These Changes
* Replaces #1900.
* Brings #1557 up to speed with dev.
* For more info, see comprehensive issue #1415

Co-authored-by: Matthew Doty <mpwd@casperlabs.io>
Co-authored-by: Austin Buddington <austin@casperlabs.io>
Co-authored-by: Austin Buddington <austin.j.buddington@gmail.com>
Co-authored-by: Fraser Hutchison <fraser@casperlabs.io>

goral09 committed:
Log at error level.

goral09 committed:
Return internal error for failed query.

Commit:
fix merge conflicts with dev.

casperlabs-bors-ng[bot] committed:
Merge #2005

2005: Make internal types in EE integration tests public. r=nilz3ro a=nilz3ro

# About These Changes
* Changed `SessionBuilder::new()` to take a `DeployItemBuilder`.
* Deleted  `account.rs`.
* Deleted `code.rs`.
* Deleted `error.rs`.
* Deleted `value.rs`.
* Renamed `WasmTestBuilder` to `WasmTestContext`.
* Deprecated `SessionBuilder`.
* Updated call sites of `Value::into_t()` to use `CLValue::into_t()`.
* Removed call sites of `WasmTestContext::finish()`. 
* closes #1978 

Co-authored-by: Austin Buddington <austin@casperlabs.io>
Co-authored-by: Austin Buddington <austin.j.buddington@gmail.com>

casperlabs-bors-ng[bot] committed:
Merge #2249

2249: Do not crash node when AuctionState query can't be answered. r=goral09 a=goral09



Co-authored-by: Mateusz Górski <gorski.mateusz@protonmail.ch>

afck committed:
Merge remote-tracking branch 'upstream/dev' into fast-sync-merge

afck committed:
Fix client no-std build.

xcthulhu commented on client/lib/validation.rs:
```diff
@@ -4,11 +4,10 @@ use jsonrpc_lite::JsonRpc;
 use thiserror::Error;
```
Might as well merge this whole file back into `dev`

xcthulhu commented on node/src/components/block_validator.rs:
```diff
@@ -203,7 +203,7 @@ where
+            peer_id=?sender, %duplicates,
             "received invalid block containing duplicated deploys"
         );
     }
```
Below in this file we should remove:

```rust
        if deploy.deploy_or_transfer_hash() != dt_hash {
            warn!(
                deploy = ?deploy,
                expected_deploy_or_transfer_hash = ?dt_hash,
                actual_deploy_or_transfer_hash = ?deploy.deploy_or_transfer_hash(),
                "Deploy has incorrect transfer hash"
            );
            return Event::CannotConvertDeploy(dt_hash);
        }
```

afck committed:
Increase timeout for test_joiner_network.

It doesn't fail locally, so probably it's just slower in CI.

afck commented:
I created a draft PR https://github.com/casper-network/casper-node/pull/2266 (that we won't merge!) which shows the diff with `dev` instead of `feat-fast-sync`.

afck committed:
Fix minor style issues.

afck committed:
Revert #1928: Make Block::verify public again.

This approach to block verification won't be used after all. We might go
with the decorator approach instead later.

afck committed:
Add newline, so rustfmt doesn't change order.

afck committed:
Revert removed newline.

afck committed:
Revert some accidental changes.

afck committed:
Address review comments.

afck committed:
Address review comments.

afck committed:
Make FetcherMetrics pub(crate).

afck committed:
Add back the deploy hash check.

This is needed after all: The fetcher won't return a deploy with the
wrong hash, but it doesn't distinguish between transfers and other
deploys.

afck committed:
Use storage integrity check code from dev.

afck committed:
Fix formatting.

afck committed:
Address comments; storage fixes.

afck committed:
Remove unneeded Clone/Serialize derives.

afck committed:
Revert API change.

afck commented:
bors merge

xcthulhu commented:
@afck I don't think we can do bors-merge because the upgrade tests are failing.

afck commented:
Right, that's https://github.com/casper-network/casper-node/issues/1665 and not due to this merge. :+1:
